### PR TITLE
feat: dashmate for hpmn

### PR DESCRIPTION
**Pull request by pshenmic**
## Issue being fixed or feature implemented

Integration of `dashmate` into deploy tool for masternode hosts

## What was done?

* masternode hosts are now deployed via `dashmate` package
* added destroy feature for both platform and network modes
* migrated all the tests to work with dashmate

## How Has This Been Tested?

Platform test suite + smoke tests

## Breaking Changes

* Merged core and platform roles into a single one
* Dashmate is now deployed on masternode hosts

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated relevant unit/integration/functional/e2e tests
- [x] I have made corresponding changes to the documentation

**For repository code-owners and collaborators only**
- [x] I have assigned this pull request to a milestone

### CODE REVIEW

**Review by shumkov — CHANGES_REQUESTED** [review]

File: ansible/roles/dashmate/defaults/main.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+---
+dashd_debug: 0
+dashd_logips: 0
+dashd_powtargetspacing: 150
+dashd_indexes: false
```
Comment (shumkov) [inline ansible/roles/dashmate/defaults/main.yml:5]: Why indices are disabled?

**Review by shumkov — CHANGES_REQUESTED** [review]

File: ansible/group_vars/all [hunk 2]
```diff
@@ -172,3 +175,13 @@ elastic_path: '{{ dashd_home }}/{{ elastic_compose_project_name }}'
 masternode_network_latency: false
 masternode_network_latency_min: 10
 masternode_network_latency_max: 100
+
+# Dashmate
+dashmate_home: /home/dashmate
```
Comment (shumkov) [inline ansible/group_vars/all:180]: Shouldn't we move those to the role vars?

File: ansible/roles/dashmate/templates/dashmate.json.j2 [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+      },
+      "platform": {
+        "enable": {% if dashmate_platform_enable %}true{% else %}false{% endif %},
+        "dapi": {
+          "envoy": {
+            "docker": {
+              "image": "{{ dapi_envoy_image }}"
+            },
+            "http": {
+              "port": 3000
```
Comment (shumkov) [inline ansible/roles/dashmate/templates/dashmate.json.j2:81]: We should use variable here

File: ansible/roles/dashmate/templates/dashmate.json.j2 [hunk 4]
```diff
@@ -0,0 +1,240 @@
...
+              "level": "{{ tenderdash_log_level }}",
+              "format": "plain"
+            },
+            "moniker": "{{ inventory_hostname }}",
+            "node": {
+              "id": "{{ node.node_key.id }}",
+              "key": "{{ node.node_key.private_key }}"
+            },
+            "genesis": {
+              "genesis_time": "2021-07-22T12:57:05.429Z",
```
Comment (shumkov) [inline ansible/roles/dashmate/templates/dashmate.json.j2:162]: We should use varaible

File: ansible/roles/dashmate/templates/dashmate.json.j2 [hunk 5]
```diff
@@ -0,0 +1,240 @@
...
+      },
+      "dashmate": {
+        "helper": {
+          "docker": {
+            "image": "dashpay/dashmate-helper:0.24-dev"
+          }
+        }
+      },
+      "externalIp": "{{ dashd_externalip }}",
+      "network": "devnet",
```
Comment (shumkov) [inline ansible/roles/dashmate/templates/dashmate.json.j2:236]: I think this should go from network name

Comment (pshenmic) [reply ansible/roles/dashmate/templates/dashmate.json.j2:236]: Isn't this a type of network? like `testnet`, `local`, or `mainnet`

**Review by shumkov — CHANGES_REQUESTED** [review]

File: ansible/roles/dashmate/templates/dashmate.json.j2 (hunk 5 above)
Comment (shumkov) [reply ansible/roles/dashmate/templates/dashmate.json.j2:236]: Correct!

File: ansible/roles/kill_platform/tasks/main.yml [hunk 6]
```diff
@@ -1,10 +1,19 @@
...
+  when: inventory_hostname in groups["hp_masternodes"]
+
 - name: Remove platform docker containers and volumes
   community.docker.docker_compose:
     project_src: '{{ mn_evo_services_path }}'
     state: absent
     remove_volumes: false
+  when:
+    - inventory_hostname not in groups["hp_masternodes"]
+    - inventory_hostname not in groups["masternodes"]
```
Comment (strophy) [inline ansible/roles/kill_platform/tasks/main.yml:16]: Should this task run on `masternodes`?

Comment (shumkov) [reply ansible/roles/kill_platform/tasks/main.yml:16]: Yes, it should

File: ansible/roles/dashmate/templates/dash.conf.j2 [hunk 7]
```diff
@@ -0,0 +1,4 @@
+# dash-cli
+rpcport={{ dashd_rpc_port }}
+rpcuser={{ dashd_rpc_user }}
+rpcpassword={{ dashd_rpc_password }}
```
Comment (strophy) [inline ansible/roles/dashmate/templates/dash.conf.j2:4]:
```suggestion
rpcpassword={{ dashd_rpc_password }}

```

**Review by strophy — APPROVED** [review]
:rocket: :rocket: :rocket: LGTM and tested on `devnet-tui`

**Review by strophy — DISMISSED** [review]

**Review by strophy — APPROVED** [review]

**Review by shumkov — APPROVED** [review]
Great! 🥇

**Pull request merged**